fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java

```diff
@@ -226,28 +231,53 @@ public OptExprBuilder visitTable(TableRelation node, ExpressionMapping context)
+            HashDistributionInfo hashDistributionInfo = (HashDistributionInfo) distributionInfo;
+            List<Column> distributedColumns = hashDistributionInfo.getDistributionColumns();
+            List<Integer> hashDistributeColumns = new ArrayList<>();
+            for (Column distributedColumn : distributedColumns) {
+                hashDistributeColumns.add(columnMetaToColRefMap.get(distributedColumn).getId());
+            }
+
+            HashDistributionDesc hashDistributionDesc =
+                    new HashDistributionDesc(hashDistributeColumns, HashDistributionDesc.SourceType.LOCAL);
+            LogicalOlapScanOperator olapScanOperator = new LogicalOlapScanOperator(node.getTable(),
+                    outputVariables,
+                    colRefToColumnMetaMap.build(),
+                    columnMetaToColRefMap,
+                    DistributionSpec.createHashDistributionSpec(hashDistributionDesc),
+                    -1,
+                    null);
             olapScanOperator.setPartitionNames(node.getPartitionNames());
             olapScanOperator.setHintsTabletIds(node.getTabletIds());
             scanOperator = olapScanOperator;
         } else if (Table.TableType.HIVE.equals(node.getTable().getType())) {
-            scanOperator = new LogicalHiveScanOperator(node.getTable(), node.getTable().getType(),
-                    outputVariables, columns.build(), columnToIds.build());
+            scanOperator = new LogicalHiveScanOperator(node.getTable(), node.getTable().getType(), outputVariables,
+                    colRefToColumnMetaMap.build(), columnMetaToColRefMap, -1, null);
         } else if (Table.TableType.SCHEMA.equals(node.getTable().getType())) {
-            scanOperator = new LogicalSchemaScanOperator(node.getTable(), outputVariables,
-                    columns.build());
+            scanOperator =
+                    new LogicalSchemaScanOperator(node.getTable(), outputVariables, colRefToColumnMetaMap.build(),
+                            columnMetaToColRefMap, -1,
+                            null);
         } else if (Table.TableType.MYSQL.equals(node.getTable().getType())) {
-            scanOperator = new LogicalMysqlScanOperator(node.getTable(), outputVariables, columns.build());
+            scanOperator =
+                    new LogicalMysqlScanOperator(node.getTable(), outputVariables, colRefToColumnMetaMap.build(),
+                            columnMetaToColRefMap, -1,
+                            null);
         } else if (Table.TableType.ELASTICSEARCH.equals(node.getTable().getType())) {
-            scanOperator = new LogicalEsScanOperator(node.getTable(), outputVariables, columns.build());
+            scanOperator =
+                    new LogicalEsScanOperator(node.getTable(), outputVariables, colRefToColumnMetaMap.build(),
+                            columnMetaToColRefMap, -1,
+                            null);
```

Comment:
1. -1 can be replace to DEFAULT_NONE_LIMIT
2. Most operator don't need set limit/predicate, canbe support a construnct method don't give limit/predicate

Reply:
1、There are too many modified codes at present, exceeding the limit of a review. I can submit another PR to modify this question
2、Can be provided, such as the first build in relationTransformer, but I do not recommend using this constructor